### ISSUE TITLE
FileBucket: Error uploading after deploying backend to heroku

### ISSUE THREAD

**Issue by kfili** [ga-wdi-boston/group-project#203]
The file upload function is not working as I am not sure how to set AWS credentials securely so I haven't done that yet. I'm also not sure how to connect my heroku deployment to mlab. I had followed all the intstructions in the express api deployment guide:
https://github.com/ga-wdi-boston/express-api-deployment-guide

For AWS credentials do I just copy those and set them in config vars in the heroku dashboard?

When uploading a file using the deployed frontend I got the following error in browser:
``` 
XHR finished loading: GET "https://sheltered-taiga-22202.herokuapp.com/categories".

2016-12-19 15:57:09.915 sheltered-taiga-22202.herokuapp.com/docs:1 POST https://sheltered-taiga-22202.herokuapp.com/docs 422 (Unprocessable Entity)
```

**Comment by raq929** [comment]
These sound like two separate issues? 

I'm going to deal with the AWS one, since it's the one in the title. Can you open a separate one about the dropdown, and explain to me how the dropdown works locally?

Yes, you can set the AWS vars on the heroku dashboard or via the command line. As long as you don't commit them to gitHub, you should be OK.

**Comment by BenGitsCode** [comment]
As Rachel said, two issues, so I'll respond to the first. 

You're certain that you've pushed your most up-to-date changes, on `master` branch to heroku?

**Comment by kfili** [comment]
@raq929 Ok thanks for confirming.

**Comment by kfili** [comment]
@BenGitsCode I did push the latest changes to master, and then to heroku
 ~~git push origin heroku~~ ``` git push heroku master ```

EDIT: fixed the typo. @payne-chris-r do I loose the thumbs down now?

**Comment by raq929** [comment]
Open a separate issue for the dropdown, please! (I have ideas, but to keep the queue useful to future cohorts, separate issues are nice.)

**Comment by kfili** [comment]
@raq929 FYI I opened another issue for the categories dropdown.

Will test upload after adding AWS credentials now.

**Comment by kfili** [comment]
@raq929 @BenGitsCode So I added my AWS creds as config vars under the same keys as in my env file.
I then ran ```heroku restart```

I'm still getting 422 (Unprocessable Entity) when I try to upload a file.

**Comment by raq929** [comment]
Fun! Check how your setting your variables locally vs on Heroku. Is there a difference? (I'm looking especially for typos in variable names.) 
This all works fine locally, correct?
Of course, don't post your keys in this issue. (I know you know this, but for future people.)

**Comment by raq929** [comment]
Also, from the command line in your repo, run `heroku config:get WHATEVER_THE_VARIABLE_NAME_IS` and make sure that it's the same on heroku as it is in your .env file.

**Comment by kfili** [comment]
Everything checks out when running ```heroku config:get``` just like in the local .env

Heres what i have on heroku dashboard:
SECRET_KEY=(different from the secret key the local repo)
CLIENT_ORIGIN=https://awesome-dream-org.github.io
CLIENT_URL=https://awesome-dream-org.github.io/TimeCapsule-Frontend
MONGODB_URI=(sent in slack to @raq929 )
AWS_S3_BUCKET_NAME
AWS_ACCESS_KEY_ID
AWS_SECRET_ACCESS_KEY

**Comment by gaand** [comment]
`git push heroku master`?

**Comment by kfili** [comment]
Hi @gaand, I re pushed to heroku master.

Still getting the same error when trying to upload: ```422 (Unprocessable Entity)```

FYI, I added two notes (in parens) about the variables in my previous comment here.

**Comment by raq929** [comment]
Can you `heroku logs` and see what it says?

**Comment by kfili** [comment]
I couldn't see any clear or enlightening errors. below are the latest bunch of logs. I will slack you all the logs so you can tell me if its safe to post them all here in this issue?

```
12-20T03:26:25.127717 app[web.1]: category is  undefined
12-20T03:26:25.137922 app[web.1]: POST /docs 422 949.751 ms - 20
12-20T03:58:57.797547 heroku[web.1]: Idling
12-20T03:58:57.798207 heroku[web.1]: State changed from up to down
12-20T03:58:58.365351 heroku[web.1]: Stopping all processes with SIGTERM
12-20T03:58:58.486226 heroku[web.1]: Process exited with status 143
```

**Comment by kfili** [comment]
Another thing I noticed earlier, is when I open ```mlab.com/home``` there are no MongoDB Deployments

**Comment by BenGitsCode** [comment]
Have you tried reading the mlab documentation for connecting to heroku?

**Comment by raq929** [comment]
What's the status of this issue?

**Comment by kfili** [comment]
Chris helped me access my remote database via mongoDB shell. 

Now I'm thinking upload does not work because there are no categories available, so this issue may be caused by issue #206 . So I'm trying to figure out how to do mongoimport to the remote database.

**Comment by kfili** [comment]
I manually added a document to our ```categories``` collection.

Now upload works because backend validation was rejecting it with a 422 error due to their being no category for the upload.
